packages/dev/core/src/FlowGraph/Blocks/Execution/flowGraphTimerBlock.ts

```diff
@@ -0,0 +1,67 @@
+import type { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+import type { FlowGraphSignalConnection } from "../../flowGraphSignalConnection";
+import { AdvancedTimer } from "../../../Misc/timer";
+import type { FlowGraphContext } from "../../flowGraphContext";
+import { FlowGraphAsyncExecutionBlock } from "../../flowGraphAsyncExecutionBlock";
+import { RichTypeNumber } from "../../flowGraphRichTypes";
+
+/**
+ * @experimental
+ * Block that can execute an action immediately and another after a delay.
+ * The delay is counted on the scene's tick.
+ */
+export class FlowGraphTimerBlock extends FlowGraphAsyncExecutionBlock {
+    public readonly timeout: FlowGraphDataConnection<number>;
+    public readonly onTimerDone: FlowGraphSignalConnection;
+
+    constructor() {
+        super();
+
+        this.timeout = this._registerDataInput("timeout", RichTypeNumber);
+        this.onTimerDone = this._registerSignalOutput("onTimerDone");
+    }
+
+    public _preparePendingTasks(context: FlowGraphContext): void {
+        const currentTimeout = this.timeout.getValue(context);
+
+        if (currentTimeout !== undefined && currentTimeout >= 0) {
+            const timers = context._getExecutionVariable(this, "runningTimers") || [];
+            const scene = context.graphVariables.scene;
+            const timer: AdvancedTimer = new AdvancedTimer({
+                timeout: currentTimeout,
+                contextObservable: scene.onBeforeRenderObservable,
+                onEnded: () => this._onEnded(timer, context),
+            });
+            timer.start();
+
+            timers.push(timer);
+            context._setExecutionVariable(this, "runningTimers", timers);
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public _execute(context: FlowGraphContext) {
+        this._startPendingTasks(context);
+        this.onDone._activateSignal(context);
+    }
+
+    private _onEnded(timer: AdvancedTimer, context: FlowGraphContext) {
+        const timers = context._getExecutionVariable(this, "runningTimers") || [];
+        const index = timers.indexOf(timer);
+        if (index !== -1) {
+            timers.splice(index, 1);
+        }

```

Comment:
is it not, technically, a bug if index is -1? should we log it?

Reply:
Hmmm yeah, it is a situation that shouldn't happen, we can log if it happens